lib/ColumnManager/readme.md

```diff
@@ -79,13 +79,56 @@ In the example below, we wrap the `<ColumnManager>` around the results pane and
     </ColumnManager>
 ```
 
+## Hooks Usage
+A typical use-case for the `<ColumnManager>` is the implementation of toggleable columns in a results view.
+
+In the example below, we use `useColumnManager` hook to manage columns visiblity and `ColumnManagerMenu` component to display controls instead of wrapping components by `ColumnManager`.
+
+```js
+    import { ColumnManagerMenu, useColumnManager } from '@folio/stripes/smart-components';
+    import { Pane, MultiColumnList } from '@folio/stripes/components';
+
+    const columnMapping = {
+        status: 'Status',
+        name: 'Name',
+        barcode: 'Barcode',
+        username: 'Username',
+        email: 'Email'
+    }
+
+    const { visibleColumns, toggleColumn } = useColumnManager('users-column-manager', columnMapping);
+
+    const actionMenu = () => (
+        <>
+            {/* Other menu sections */}
+            <ColumnManagerMenu
+                prefix="users"
+                columnMapping={columnMapping}
+                visibleColumns={visibleColumns}
+                toggleColumn={toggleColumn}
+            />
+        </>
+    );
```

Comment:
Can I propose the API
```suggestion
    const columnManager = useColumnManager({ prefix: 'users', columnMapping });

    const actionMenu = () => (
        <>
            {/* Other menu sections */}
            <ColumnManagerMenu {...columnManager} />
        </>
    );
```

Reply:
props are different with return value

Reply:
Yes, the point is to adjust the return value too.
So we could use `prefix` and `columnMapping` just once as initializing values.

Reply:
1. prefix values are different
2. why should we return the same object from args when it can be used without? related to `columnMapping`
3. `columnManager` is a new object each render (at least with current implementation) and useMemo/useCallback won't make sense with such usage

Reply:
Yes, we could use define `storageKey` as `${prefix}-columns-storage` in the hook (instead of just `${prefix}-storage`. The benefit is having just one prefix.

The point of returning the same `columnMapping` is just to make API surface cleaner - one entry point for all init is the hook.
And I wouldn't worry about `columnManager` being a new object. It's a common pattern for hooks, which is processed by React just fine, I believe.

Reply:
`It's a common pattern for hooks` - destructing is used mostly everywhere, with it you suggestion doesn't make sense

otherwise memoized object is returned

Reply:
For my suggestion I kept in mind the FinalForm `useField` API, where the whole bunch of `input` props is passed - and easily used - as an object:

```js
const { input } = useField(...);
//...
<TextField {...input} />
```

I also dare say there's maybe no need in memoizing the returned object.
At least there's not much of the object memoization here, only the individual useCallbacks:
https://github.com/final-form/react-final-form/blob/cf64402a407ee08264c2348a035ddb718a058082/src/useField.js#L133-L243

Reply:
Or maybe, even more bold proposal, the "[hook returning component](https://dev.to/droopytersen/new-react-hooks-pattern-return-a-component-31bh)" pattern you mentioned recently is suitable here:
```js
const ColumnManager = useColumnManager({ prefix: 'users', columnMapping });

const actionMenu = () => (
  <>
    {/* Other menu sections */}
    <ColumnManager.Menu />
  </>

//..

<MultiColumnList
  id="list-users"
  visibleColumns={ColumnManager.columns}
  //...
/>
```